### PR TITLE
Hide/Show register button FRONT

### PR DESCRIPTION
**Pull request by PedroR-dev**
Create logic to hide or show register button based on api flag. Update RegisterController to prevent sign up if button is hidden.
Limited permissions so that the admin can only modify this field, but not all domain config. Created new POE entries

### CODE REVIEW

File: grails-app/controllers/springSecurity/RegisterController.groovy [hunk 1]
```diff
@@ -59,6 +57,8 @@ class RegisterController extends grails.plugin.springsecurity.ui.RegisterControl
         copy.remove 'action'
         def config = SpringSecurityUtils.securityConfig
 
+        registerButtonDisabledRedirect();
```
Comment (iduetxe) [inline grails-app/controllers/springSecurity/RegisterController.groovy]: Rename to checkRegisterActiveAndRedirect

File: src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy [hunk 2]
```diff
@@ -20,15 +20,16 @@ class DomainValidationCommand {
...
     Boolean providerBasicEmailForm
     Boolean providerGoogle
     Boolean providerFacebook
     Boolean providerAoc
 
     static constraints = {
-        domainPrivacy nullable: false
-        firstFactorValidation nullable: false
+        domainPrivacy nullable: true
+        firstFactorValidation nullable: true
```
Comment (iduetxe) [inline src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy]: REvert

File: src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy [hunk 3]
```diff
@@ -20,15 +20,16 @@ class DomainValidationCommand {
     Boolean isSocialNetwork
     Boolean isUserProfileExtended
     Boolean tourEnabled
+    Boolean showRegisterButton
```
Comment (iduetxe) [inline src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy]: Revert. As we talked this field will be in other command

**Review by iduetxe — CHANGES_REQUESTED** [review]

**Comment by PedroR-dev** [comment]
A futuro cuando se pueda me encantaría juntar la lógica de los dos checkbox, creo que no tiene sentido que estén separados

**Review by iduetxe — APPROVED** [review]

**Pull request merged**